rules/Php84/Rector/Class_/DeprecatedAnnotationToDeprecatedAttributeRector.php

```diff
@@ -0,0 +1,178 @@
+/**
+ * @deprecated 1.0.0 Use SomeOtherClass instead
+ */
+class SomeClass
+{
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+#[\Deprecated(message: 'Use SomeOtherClass instead', since: '1.0.0')]
+class SomeClass
+{
+}
+CODE_SAMPLE
+            ),
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+/**
+ * @deprecated 1.0.0 Use SomeOtherFunction instead
+ */
+function someFunction()
+{
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+#[\Deprecated(message: 'Use SomeOtherFunction instead', since: '1.0.0')]
+function someFunction()
+{
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [
+            Class_::class,
+            Interface_::class,
+            Trait_::class,
+            Function_::class,
+            ClassMethod::class,
+            Property::class,
+            ClassConst::class,
+        ];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        if (
+            ! $node instanceof ClassConst
+            && ! $node instanceof ClassMethod
+            && ! $node instanceof Property
+            && ! $node instanceof Class_
+            && ! $node instanceof Interface_
+            && ! $node instanceof Trait_
+            && ! $node instanceof Function_
+        ) {
+            return null;
+        }
+
+        $hasChanged = false;
+        $phpDocInfo = $this->phpDocInfoFactory->createFromNode($node);
+        if ($phpDocInfo instanceof PhpDocInfo) {
+            $requiresAttributeGroups = $this->handleDeprecated($phpDocInfo);
+            if ($requiresAttributeGroups !== []) {
+                $this->docBlockUpdater->updateRefactoredNodeWithPhpDocInfo($node);
+                $node->attrGroups = array_merge($node->attrGroups, $requiresAttributeGroups);
+                $this->removeDeprecatedAnnotations($phpDocInfo);
+                $hasChanged = true;
+            }
+        }
+
+        return $hasChanged ? $node : null;
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::DEPRECATED_ATTRIBUTE;
+    }
+
+    /**
+     * @return list<AttributeGroup>
+     */
+    private function handleDeprecated(PhpDocInfo $phpDocInfo): array
+    {
+        $attributeGroups = [];
+        $desiredTagValueNodes = $phpDocInfo->getTagsByName('deprecated');
+        foreach ($desiredTagValueNodes as $desiredTagValueNode) {
+            if (! $desiredTagValueNode->value instanceof DeprecatedTagValueNode) {
+                continue;
+            }
+
+            $attributeGroups[0] = $this->createAttributeGroup($desiredTagValueNode->value->description);
+            $this->phpDocTagRemover->removeTagValueFromNode($phpDocInfo, $desiredTagValueNode);
+        }
+
+        return $attributeGroups;
+    }
+
+    private function createAttributeGroup(string $annotationValue): AttributeGroup
+    {
+        $pattern = '/^(?:(\d+\.\d+\.\d+)\s+)?(.*)$/';
```

Comment:
please move to constant, and add `@see https://regex101...` url to show the regex example